src/parser.nim

```diff
@@ -87,10 +87,16 @@ proc parseVideo(js: JsonNode): Video =
     result.description = description.getStr
 
   for v in js{"video_info", "variants"}:
+    let contentType = parseEnum[VideoType](v{"content_type"}.getStr("summary"))
+    let url = v{"url"}.getStr
+    let resolution =
+      if contentType == mp4 and url =~ re".*/vid/\d+x(\d+)/": parseInt(matches[0])
```

Comment:
This could be turned into a `getMp4Resolution` proc in `parserutils.nim`, using a few `find` calls instead of regex, making it orders of magnitude more efficient. Here's an equivalent function tested with a few URLs:

```nim
proc getMp4Resolution(url: string): int =
  const vidSep = "/vid/"
  let
    vidIdx = url.find(vidSep) + vidSep.len
    resIdx = url.find('x', vidIdx) + 1
    res = url[resIdx ..< url.find("/", resIdx)]

  if res.len > 0:
    return parseInt(res)
```